Automattic/newspack-network
# feat(content-distribution): handle status changes

Handle content distribution status changes and deletions.

When a distributed post gets its status changed to non-publish (`draft`, `trash`, `private`, or `pending`), its linked posts should follow.

When a distributed post gets deleted, a linked post should move to the trash and the unlinked post remain with its status. The distributed post payload meta will be deleted as a way to treat the post as a regular standalone post. The network post ID and unlinked meta also remain so if the original post gets restored from a backup, its relationship in the network is also restored.

### Testing

1. Make sure you have a network setup with at least a hub and 2 nodes (let's call node1 and node2)
2. Create a post in the hub and distribute:

```php
$post_id = 0; // Change to your post ID.
$outgoing_post = new Newspack_Network\Content_Distribution\Outgoing_Post( $post_id );
$outgoing_post->set_config( [ 'http://node1.local', 'http://node2.local' ] ); // Change to match your nodes.
Newspack_Network\Content_Distribution::distribute_post( $outgoing_post );
```

3. Once it is distributed, publish the post in node1 and node2
4. In node2, unlink the post via shell:

```php
$post_id = 0; // Change to the post ID in node2.
$incoming_post = new Newspack_Network\Content_Distribution\Linked_Post( $post_id );
$incoming_post->set_unlinked();
```

5. In the hub, change the post status to `draft`
6. Confirm it moves to draft in node1 and remains published in node2
7. In the hub, delete the post
8. Confirm it moves to the trash in node1 and remains published in node2
9. In both node1 and node2, confirm the post payload meta is gone:

```php
get_post_meta( $post_id, 'newspack_network_post_payload', true );
```
